### ISSUE TITLE
Addition of maxEpochs to Early Stop at Min Loss

### ISSUE THREAD

**Issue by kartikdutt18** [mlpack/ensmallen#173]
Hi everyone, I was training an LSTM using mlpack. A portion of the code is given below:

```
    optimizer.Tolerance() = -1;
    optimizer.MaxIterations() = 0;

    model.Train(trainX,
                trainY,
                optimizer,
                ens::PrintLoss(),
                ens::ProgressBar(),
                ens::EarlyStopAtMinLoss(100),
                ens::StoreBestCoordinates<arma::mat>());
```
The model converges at a local minima if patience is very but overfits with a patience of 100. I was wondering if there was a way to terminate it after say 100th epoch irrespective of whether it reached a minima.
This way we could avoid using `for loops` altogether.
I was trying to find a solution such that I could use EarlyStopAtMinLoss and not exceed 100 epochs and not use `for loops`.
One solution would be to add a condition in the EndEpoch.
I am not sure if this necessary, So I would love to hear your opinion.
If a way of doing the same doesn't exist, can I implement the same.
Sorry if I missed something.
Thanks.

**Comment by rcurtin** [comment]
Hmm, could you set `optimizer.MaxIterations()` to `trainX.n_cols * 100`?  Or maybe I overlooked something?

**Comment by shrit** [comment]
Interesting, I think you are using `ens::EarlyStopAtMinLoss(100)` on the training set, and not on the validation set, If it is the case, I think this is the reason for overfitting in this case. We have discussed this is in https://github.com/mlpack/mlpack/issues/2185 and I have an open pull request in https://github.com/mlpack/ensmallen/pull/165

However, I think fixing the size of epochs is a good idea, but we can pass it directly in the optimizer parameters through the `MaxIteration()`

**Comment by kartikdutt18** [comment]
>  I think you are using ens::EarlyStopAtMinLoss(100) on the training set,

Yes, I was using it on training set that's why it was overfitting.

> Hmm, could you set optimizer.MaxIterations() to trainX.n_cols * 100? Or maybe I overlooked something?

Sorry I missed such an obvious thing and thanks for quick reply. 
This really helped so I'm closing this issue.
Thanks.
